includes/classes/Indexable/Term/SyncManager.php

```diff
@@ -185,10 +203,6 @@ public function action_sync_on_delete( $term_id ) {
 		$hierarchy = array_merge( $ancestors, $children );
 
 		foreach ( $hierarchy as $hierarchy_term_id ) {
-			if ( ! current_user_can( 'edit_term', $hierarchy_term_id ) ) {
```

Comment:
What are your thoughts on checking if `ep_sync_delete_permissions_bypass` is applied as well instead of completely removing the permissions check? I'm seeing that it's hooked onto `filter_bypass_permission_checks_for_machines` to account for CLI: https://github.com/10up/ElasticPress/blob/e3639bbcdec9efdc4680162563bc215c85c6e6bb/includes/classes/Indexable/Post/SyncManager.php#L68.

Reply:
Yeah there is `ep_sync_delete_permissions_bypass` as well. I would say it is a plan B on push back :) But I don't really see the benefit of doing permission checking at this state....oooh now I do. I guess anybody's evil plugin could invoke the `delete_term` hooks not just core WordPress? So even if we are not really deleting the term.

But then again, we are not deleting anything here. We are just saying please update these items to what they really are in DB.

Reply:
I re-added the check now. Thanks for the input.